pkg/backporter/backporter.go

```diff
@@ -0,0 +1,296 @@
+	<p> Target Release: {{ .Bug.TargetRelease }} </p>
+	{{ if .PRs }}
+		<p> GitHub PR: 
+		{{ if .PRs }}
+			{{ range $index, $pr := .PRs }}
+				{{ if $index}}|{{end}} 
+				<a href={{ $pr.Type.URL }}/{{ $pr.Org}}/{{ $pr.Repo}}/pull/{{ $pr.Num}}>{{ $pr.Org}}/{{ $pr.Repo}}#{{ $pr.Num}}</a>
+			{{ end }}
+		{{ end }}
+		</p>
+	{{ else }}
+		<p> No linked PRs! </p>
+	{{ end }}
+	{{ if ne .Parent.ID .Bug.ID}}
+		<p> Cloned From: <a href = /getclones?ID={{.Parent.ID}}> Bug {{.Parent.ID}}: {{.Parent.Summary}}</a> | Status: {{.Parent.Status}}
+	{{ else }}
+		<p> Cloned From: This is the original! </p>
+	{{ end }}
+	<h4 id="clones"> <a href ="#clones"> Clones</a> </h4>
+	<table class="table">
+		<thead>
+			<tr>
+				<th title="Targeted version to release fix" class="info">Target Release</th>
+				<th title="ID of the cloned bug" class="info">Bug ID</th>
+				<th title="Status of the cloned bug" class="info">Status</th>
+				<th title="PR associated with this bug" class="info">PR</th>
+			</tr>
+		</thead>
+		<tbody>
+		{{ if .Clones }}
+			{{ range $clone := .Clones }}
+				<tr>
+					<td style="vertical-align: middle;">{{ $clone.TargetRelease }}</td>
+					<td style="vertical-align: middle;"><a href = /getclones?ID={{$clone.ID}}>{{ $clone.ID }}</a></td>
+					<td style="vertical-align: middle;">{{ $clone.Status }}</td>
+					<td style="vertical-align: middle;">
+						{{range $index, $pr := $clone.PRs }}
+							{{ if $index}},{{end}}
+							<a href = {{ $pr.Type.URL }}/{{$pr.Org}}/{{$pr.Repo}}/pull/{{$pr.Num}} target="_blank"> {{$pr.Org}}/{{$pr.Repo}}#{{$pr.Num}}</a>
+						{{end}}
+					</td>
+				</tr>
+			{{ end }}
+		{{ else }}
+			<tr> <td colspan=4 style="text-align:center;"> No clones found! </td></tr>
+		{{ end }}
+		</tbody>
+	</table>`
+
+var (
+	clonesTemplate = template.Must(template.New("clones").Parse(clonesTemplateConstructor))
+	emptyTemplate  = template.Must(template.New("empty").Parse(emptyTemplateConstructor))
+)
+
+// HandlerFuncWithErrorReturn allows returning errors to be logged
+type HandlerFuncWithErrorReturn func(http.ResponseWriter, *http.Request) error
+
+type wrapper struct {
+	Bug    *bugzilla.Bug
+	Clones []*bugzilla.Bug
+	Parent *bugzilla.Bug
+	PRs    []bugzilla.ExternalBug
+}
+
+//Writes an HTML page, prepends header in htmlPageStart and appends header from htmlPageEnd around tConstructor.
+func writePage(w http.ResponseWriter, title string, body *template.Template, data interface{}) error {
+	fmt.Fprintf(w, htmlPageStart, title)
+
+	if err := body.Execute(w, data); err != nil {
+		w.WriteHeader(http.StatusInternalServerError)
+		fmt.Fprintf(w, "%s: %v", http.StatusText(http.StatusInternalServerError), err)
+		return err
+	}
+	fmt.Fprint(w, htmlPageEnd)
+	return nil
+}
+
+// GetLandingHandler will return a simple bug search page
+func GetLandingHandler() HandlerFuncWithErrorReturn {
+	return func(w http.ResponseWriter, req *http.Request) error {
+		writePage(w, "Home", emptyTemplate, nil)
```

Comment:
Err, that is a very complicated way to print a string into the body. Simpler would be:

```
if _, err := w.Write([]byte("Home")); err != nil {
  return err
}
```

(tbh I do not see a god reason for `emptyTemplate` to exist. Just write directly where its currently used. )

Reply:
`fmt.Fprintf(w, "format error: %v", err)` also works for simple cases

Reply:
The error needs to go in the page and dont think it would be good aesthetically to send just the error message, especially considering that this is not an ajax call but a full page render.